src/commands/deployment/self-test.ts

```diff
@@ -0,0 +1,177 @@
+      config.introspection!.azure!.partition_key!,
+      config.introspection!.azure!.table_name!,
+      buildId
+    );
+
+    const statusMessage =
+      "Finished running self-test. Service introspection self-test status: ";
+
+    if (!isVerified) {
+      logger.error(statusMessage + "FAILED. Please try again.");
+    } else {
+      logger.info(statusMessage + "SUCCEEDED.");
+    }
+  } catch (err) {
+    logger.error("Error running self-test.");
+    logger.error(err);
+  }
+};
+
+/**
+ * Writes self-test pipeline data to the storage account table.
+ * @param accountKey storage account key
+ * @param accountName storage account name
+ * @param partitionKey storage account table partition key
+ * @param tableName storage account table name
+ */
+export const writeSelfTestData = async (
+  accountKey: string,
+  accountName: string,
+  partitionKey: string,
+  tableName: string
+): Promise<string> => {
+  // call create
+  const tableInfo: IDeploymentTable = {
+    accountKey,
+    accountName,
+    partitionKey,
+    tableName
+  };
+
+  const buildId = Math.floor(Math.random() * 1000).toString();
+
+  try {
+    const p1Id = buildId;
+    const imageTag = "spk-test-123";
+    const commitId = "6nbe";
+    const env = "SPK-TEST";
+
+    logger.info("Adding src to ACR data to service introspection...");
+    await addSrcToACRPipeline(tableInfo, p1Id, imageTag, service, commitId);
+
+    const p2Id = buildId;
+    logger.info("Adding ACR to HLD data to service introspection...");
+    await updateACRToHLDPipeline(tableInfo, p2Id, imageTag, commitId, env);
+
+    return new Promise<string>(resolve => {
+      resolve(buildId);
+    });
+  } catch (err) {
+    logger.error("Error writing data to service introspection.");
+    logger.error(err);
+
+    return new Promise<string>(resolve => {
+      resolve("");
+    });
+  }
+};
+
+/**
+ * Deletes self-test pipeline data from the storage account table.
+ * self-test data is identified by the 'self-test' service name.
+ * @param accountKey storage account key
+ * @param accountName storage account name
+ * @param partitionKey storage account table partition key
+ * @param tableName storage account table name
+ * @param buildId build ID for the test data that was added
+ */
+export const deleteSelfTestData = async (
+  accountKey: string,
+  accountName: string,
+  partitionKey: string,
+  tableName: string,
+  buildId: string
+): Promise<boolean> => {
+  // search by service
+  const tableInfo: IDeploymentTable = {
+    accountKey,
+    accountName,
+    partitionKey,
+    tableName
+  };
+
+  const isDeleted = await findMatchingDeployments(
+    tableInfo,
+    "service",
+    service
+  ).then(async entries => {
+    let entryToDelete: any;
+    logger.info("Deleting test data...");
+    let foundEntry = false;
+
+    try {
+      for (const entry of entries) {
+        entryToDelete = entry;
+
+        if (entry.p1 && entry.p1._ === buildId) {
```

Comment:
nit: I'm thinking if we should use something other than a buildId for this verification (writeSelfTestData should perhaps return the RowKey instead of the buildId) in case there's two entries in the table with this build Id?

Reply:
Do you think that's needed even though this is already only fetching the entries for the `self-test` service?

Reply:
@samiyaakhtar also the buildId is random

Reply:
@edaena oh yeah then it should be fine